### PR TITLE
feat: adding spot arazzo rules

### PR DESCRIPTION
**Pull request by DmitryAnansky**
## What/Why/How?

Adding Spot Arazzo rules and docs:

- [x]  Unique criteria rule is missing
- [x]  not suport actions type end
- [x]  No support for XPath.

## Check yourself

- [ ] Code changed? - Tested with redoc/reference-docs/workflows (internal)
- [x] All new/updated code is covered with tests
- [ ] New package installed? - Tested in different environments (browser/node)

## Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### CODE REVIEW

File: resources/arazzo.yaml [hunk 1]
```diff
@@ -68,7 +68,9 @@ workflows:
           - condition: $statusCode == 201
           - context: $response.body
             condition: $.name == 'Mermaid Treasure Identification and Analysis'
-            type: jsonpath
+            type:
```
Comment (DmitryAnansky) [inline resources/arazzo.yaml:71]: Just to illustrate possible option.

File: .changeset/green-penguins-press.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+"@redocly/openapi-core": minor
+"@redocly/cli": minor
+---
+
+Add Spot and Arazzo rules: 'no-criteria-xpath', 'no-actions-type-end', 'criteria-unique'.
```
Comment (JLekawa) [inline .changeset/green-penguins-press.md:6]:
```suggestion
Added Spot and Arazzo rules: `no-criteria-xpath`, `no-actions-type-end`, `criteria-unique`.
```

File: docs/guides/lint-arazzo.md [hunk 3]
```diff
@@ -84,6 +84,9 @@ The rules available for linting Arazzo are:
 - `step-onSuccess-unique`: the `onSuccess` actions of the `step` object must be unique.
 - `step-onFailure-unique`: the `onFailure` actions of the `step` object must be unique.
 - `requestBody-replacements-unique`: the `replacements` of the `requestBody` object must be unique.
+- `no-criteria-xpath`: the `xpath` type criteria is not supported by Spot.
+- `no-actions-type-end`: the `end` type action is not supported by Spot.
+- `criteria-unique`: the criteria list should not contain duplicated assertions.
```
Comment (JLekawa) [inline docs/guides/lint-arazzo.md:89]:
```suggestion
- `criteria-unique`: the criteria list must not contain duplicated assertions.
```

File: docs/rules/arazzo/criteria-unique.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
+---
+slug: /docs/cli/rules/arazzo/criteria-unique
+---
+
+# criteria-unique
+
+The criteria list should not contain duplicated assertions.
```
Comment (JLekawa) [inline docs/rules/arazzo/criteria-unique.md:7]:
```suggestion
The criteria list must not contain duplicated assertions.
```

File: docs/rules/arazzo/criteria-unique.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+| Arazzo | Compatibility |
+| ------ | ------------- |
+| 1.0.0  | ✅            |
+
+## API design principles
+
+The criteria list should not contain duplicated assertions.
```
Comment (JLekawa) [inline docs/rules/arazzo/criteria-unique.md:15]:
```suggestion
The criteria list must not contain duplicated assertions.
```

File: docs/rules/spot/no-actions-type-end.test.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+
+The `end` type action is not supported by Spot.
+
+| Arazzo | Compatibility |
+| ------ | ------------- |
+| 1.0.0  | ✅            |
+
+## API design principles
+
+This is `Spot` specific rule.
```
Comment (JLekawa) [inline docs/rules/spot/no-actions-type-end.test.md:15]:
```suggestion
This is a `Spot`-specific rule.
```

File: docs/rules/arazzo/criteria-unique.md [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+      - condition: $response.body.open == true
+      - condition: $response.body.open == true
+onFailure:
+  - name: 'onFailureActionName'
+    type: 'goto'
+    stepId: 'buy-ticket'
+    criteria:
+      - condition: $response.body.open == true
+      - condition: $response.body.open == true
+```
```
Comment (JLekawa) [inline docs/rules/arazzo/criteria-unique.md:59]: I'd remove all examples of incorrect criteria - usually users are not reading documentation from top to bottom, or are in a hurry and will copy the wrong example. without noticing.

File: docs/rules/arazzo/criteria-unique.md [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+
+Example of a **correct** criteria:
```
Comment (JLekawa) [inline docs/rules/arazzo/criteria-unique.md:61]:
```suggestion
Example of **correct** criteria:
```

File: docs/rules/spot/no-actions-type-end.test.md [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+```yaml Object example
+onSuccess:
+  - name: 'onSuccessActionName'
+    type: 'end'
+    stepId: 'buy-ticket'
+onFailure:
+  - name: 'onFailureActionName'
+    type: 'end'
+    stepId: 'buy-ticket'
+```
```
Comment (JLekawa) [inline docs/rules/spot/no-actions-type-end.test.md:51]: See my previous comment on wrong examples.

File: docs/rules/spot/no-criteria-xpath.md [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+      version: draft-goessner-dispatch-jsonpath-00
+  - context: $response.body
+    condition: /bookstore/book[price>35]/price
+    type: xpath
+  - context: $response.body
+    condition: /bookstore/book[price>35]/price
+    type:
+      type: xpath
+      version: xpath-30
+```
```
Comment (JLekawa) [inline docs/rules/spot/no-criteria-xpath.md:58]: See my previous comment on wrong examples.

File: docs/rules/spot/no-actions-type-end.test.md [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+
+Example of a **correct** action:
```
Comment (JLekawa) [inline docs/rules/spot/no-actions-type-end.test.md:53]:
```suggestion
Example of an action:
```

File: docs/rules/spot/no-criteria-xpath.md [hunk 12]
```diff
@@ -0,0 +1,74 @@
...
+
+Example of a **correct** criteria:
```
Comment (JLekawa) [inline docs/rules/spot/no-criteria-xpath.md:60]:
```suggestion
Example of criteria:
```

File: docs/rules/arazzo/criteria-unique.md (hunk 7 above)
Comment (DmitryAnansky) [reply docs/rules/arazzo/criteria-unique.md:59]: Sure, you can change docs format as you prefer.
Just wanted to keep rules described in the docs.

Comment (JLekawa) [inline docs/rules/arazzo/criteria-unique.md:59]:
```suggestion
```

File: docs/rules/spot/no-actions-type-end.test.md (hunk 9 above)
Comment (JLekawa) [inline docs/rules/spot/no-actions-type-end.test.md:51]:
```suggestion
```

File: docs/rules/spot/no-criteria-xpath.md (hunk 10 above)
Comment (JLekawa) [inline docs/rules/spot/no-criteria-xpath.md:58]:
```suggestion
```

**Review by JLekawa — APPROVED** [review]

**Review by tatomyr — APPROVED** [review]

**Pull request merged**